tests/test_cleanup.py

```diff
@@ -487,3 +507,884 @@ def test_file_vdi_delete(self, mock_lock):
 
         vdi.delete()
         mock_lock.Lock.cleanupAll.assert_called_with(str(vdi_uuid))
+
+    def test_coalesceLeaf(self):
+
+        sr_uuid = uuid4()
+        sr = create_cleanup_sr(uuid=str(sr_uuid))
+        vdi_uuid = uuid4()
+        vdi = cleanup.VDI(sr, str(vdi_uuid), False)
+
+        vdi.canLiveCoalesce = mock.MagicMock(autospec=True)
```

Comment:
Do you not have to tell this what the spec is for autospec to work? Or is it magic when mocking a function?

Reply:
Your right, the test won't catch someone calling the function with a wrong signature which was my intention. Annoying that python let me do this.

Mocking the function with a patch and autospec=true does catch it, 

I will refactor the tests to use patch.

Reply:
Test refactoring finally done.